packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -54,36 +60,33 @@ export class Communicator {
       window.addEventListener('message', listener);
       this.listeners.set(listener, { reject });
     });
-  }
-
-  private listeners = new Map<(_: MessageEvent) => void, { reject: (_: Error) => void }>();
+  };
 
   /**
-   * Rejects all requests and clears the listeners
+   * Closes the popup, rejects all requests and clears the listeners
    */
-  disconnect() {
+  disconnect = () => {
+    closePopup(this.popup);
+    this.popup = null;
+
     this.listeners.forEach(({ reject }, listener) => {
       reject(standardErrors.provider.userRejectedRequest('Request rejected'));
       window.removeEventListener('message', listener);
     });
     this.listeners.clear();
-  }
-
-  private popup: Window | null = null;
+  };
 
   /**
    * Waits for the popup window to fully load and then sends a version message.
    */
-  private async waitForPopupLoaded(): Promise<Window> {
+  private waitForPopupLoaded = async (): Promise<Window> => {
     if (this.popup) return this.popup;
 
     this.popup = openPopup(this.url);
 
-    this.onMessage<ConfigMessage>(({ event }) => event === 'PopupUnload').then(() => {
-      closePopup(this.popup);
-      this.popup = null;
-      this.disconnect();
-    });
+    this.onMessage<ConfigMessage>(({ event }) => event === 'PopupUnload')
```

Comment:
Discussed with @fan-zhang-sv @bangtoven 

`#disconnect` is called when there is a `PopupUnload` event, which is fired by the frontend when the window closes. Since it has been in the codebase for a while and is stable, let's not make changes until further investigation.